app/src/org/commcare/views/widgets/CalendarFragment.java

```diff
@@ -0,0 +1,256 @@
+    private void initDisplay(){
+        myGrid = (GridView) myLayout.findViewById(R.id.calendar_grid);
+        decMonth = (ImageButton) myLayout.findViewById(R.id.prevmonthbutton);
+        incMonth = (ImageButton) myLayout.findViewById(R.id.nextmonthbutton);
+        myMonth = (TextView) myLayout.findViewById(R.id.currentmonth);
+        decYear = (ImageButton) myLayout.findViewById(R.id.prevyearbutton);
+        incYear = (ImageButton) myLayout.findViewById(R.id.nextyearbutton);
+        myYear = (TextView) myLayout.findViewById(R.id.currentyear);
+    }
+
+    private void initOnClick(){
+
+        decMonth.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                myCal.add(Calendar.MONTH, -1);
+                refresh();
+            }
+        });
+
+        incMonth.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                myCal.add(Calendar.MONTH, 1);
+                refresh();
+            }
+        });
+
+        decYear.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                myCal.add(Calendar.YEAR, -1);
+                refresh();
+            }
+        });
+
+        incYear.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                myCal.add(Calendar.YEAR, 1);
+                refresh();
+            }
+        });
+
+        myGrid.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                Date date = (Date) parent.getItemAtPosition(position);
+                myCal.setTime(date);
+                refresh();
+            }
+        });
+
+        ImageButton closer = (ImageButton) myLayout.findViewById(R.id.closecalendar);
+        closer.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                dismiss();
+            }
+        });
+    }
+
+    public void refresh(){
+
+        ArrayList<Date> dateList = new ArrayList<>();
+        Calendar populator = (Calendar) myCal.clone();
+
+        int totalDays = populator.getActualMaximum(Calendar.DAY_OF_MONTH);
+
+        populator.set(Calendar.DAY_OF_MONTH, 1);
+
+        //Day of week for the first of the month
+        int monthStartWeekDay = populator.get(Calendar.DAY_OF_WEEK) - 1;
+
+        totalDays += monthStartWeekDay;
+
+        //Backtracking calendar to the most recent Sunday
+        populator.add(Calendar.DAY_OF_MONTH, -monthStartWeekDay);
+
+        while(dateList.size() < totalDays){
+            dateList.add(populator.getTime());
+            populator.add(Calendar.DAY_OF_MONTH, 1);
+        }
+
+        int remainingDays = (8-populator.get(Calendar.DAY_OF_WEEK))%7;

```

Comment:
make the numbers you're using here static final class variables with a descriptive name